tests/foreman/api/test_registration.py

```diff
@@ -262,3 +262,62 @@ def test_negative_capsule_without_registration_enabled(
         "Proxy lacks one of the following features: 'Registration', 'Templates'"
         in context.value.response.text
     )
+
+
+@pytest.mark.rhel_ver_match('[^6]')
+def test_positive_host_registration_with_non_admin_user(
+    module_org,
+    module_location,
+    module_activation_key,
+    module_target_sat,
+    rhel_contenthost,
+):
+    """Verify host registration with non admin user
+
+    :id: 02bdda6a-010d-4098-a7e0-e4b5e8416ce3
+
+    :steps:
+        1. Enable the required repositories on the satellite and get a host(RHEL7/RHEL8/RHEL9)
+        2. Create a non-admin user and assign "Register Hosts" role to it.
+        3. Create an activation key and assign CV and LCE to it.
+        4. Login Satellite with new user and generate curl command to register host with all setup options set to NO.
+        5. Run the curl command on the host.
+
+    :expectedresults: Host registered successfully with all setup options set to 'NO' for non admin user
+
+    :BZ: 2211484
+
+    :customerscenario: true
+    """
+    register_host_role = module_target_sat.api.Role().search(
+        query={'search': 'name="Register hosts"'}
+    )
+    module_target_sat.api.User(
+        role=register_host_role,
+        admin=False,
+        login=gen_string('alpha'),
+        password=gen_string('alpha'),
+        organization=[module_org],
+        location=[module_location],
+    ).create()
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_org,
+        activation_keys=[module_activation_key.name],
+        location=module_location,
+        setup_insights=False,
+        setup_remote_execution=False,
+        setup_remote_execution_pull=False,
+        update_packages=False,
+    ).create()
+    result = rhel_contenthost.execute(command)
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    # verify the setup which should not run when setup is false
+    assert_info = [
+        'dnf -y install insights-client',
+        'dnf -y install foreman_ygg_worker',
+        '# Updating packages',
+    ]
+    for data in assert_info:
+        assert data not in result.stdout
+    rendered = rhel_contenthost.execute('cat ~/.ssh/authorized_keys')
+    assert rendered.status == 1, f'{rendered.stderr}'
```

Comment:
This is potentially dangerous when the `authorized_keys` file already exists on the host. This file is on non-container hosts (VMs) where `sshd` is running. I would suggest checking if the `authorized_keys` file exists, if it does, compute its file digest before the registration is performed. After the registration, I would do the same - if the file exists and if it does not have the same digest as it had before registration (the file contents changed), I would throw an AssertionError.

Reply:
If setup is false it does not create `authorized_keys` after registration. I have make changes in AssertionError. Please check.